src/v/hashing/xx.h

```diff
@@ -39,11 +38,21 @@ class incremental_xxhash64 {
         XXH64_update(&_state, src, sz);
     }
 
-    void update(const ss::sstring& str) { update(str.data(), str.size()); }
+    // string override
+    void update(std::string_view str) { update(str.data(), str.size()); }
+
+    // named type override
+    template<
+      typename T,
+      typename std::enable_if_t<
+        std::is_convertible_v<T, typename T::type>>* = nullptr>
+    void update(const T& named_type) {
+        update(named_type());
```

Comment:
would it be possible to either add the static assert or add an additional condition to the sfinae that uses is_base_of? that would effectively be the same as the concept is_derived_from. i think this is ok, but in principle, other types thathave ::type and operator() could be here. unlikely, tho.

static_assert(std::is_base_of_v<T, named_type...> == true)

Reply:
good idea, will add this